src/test/Spec.hs

```diff
@@ -38,10 +39,15 @@ configTests :: TestTree
 configTests = testGroup "Configuration tests" $
   [ testCase file $ void $ parseConfig file | file <- files ] ++
   [ testCase "parse \"coverage: true\"" $ do
-      config <- parseConfig "coverage/test.yaml"
+      config <- _econfig <$> parseConfig "coverage/test.yaml"
       assertCoverage config $ Just mempty
   , testCase "coverage disabled by default" $
       assertCoverage defaultConfig Nothing
+  , testCase "defaults.yaml" $ do
+      EConfigWithUsage _ bad unset <- parseConfig "basic/default.yaml"
+      assertBool ("unused options: " ++ show bad) $ null bad
+      let unset' = unset `HS.difference` HS.fromList ["seed"]
+      assertBool ("unset options: " ++ show unset') $ null unset'
```

Comment:
this test case should fix our woes with people forgetting to modify `default.yaml`